lib/http/endpoint.js

```diff
@@ -190,16 +190,19 @@ const endpointBase = ({ preprocessor = noop, before = noop, resultWriter, errorW
 
 const streamErrorHandler = (response, next) => (err) => {
   response.addTrailers({ Status: 'Error' }); // TODO: improve response content.
-  response.end();
   next(err);
 };
 const defaultResultWriter = (result, request, response, next) => {
   if (!response.hasHeader('Content-Type')) response.type('json');
 
   if (result instanceof PartialPipe) {
-    result.with(response).pipeline(streamErrorHandler(response, next));
+    const errHandler = streamErrorHandler(response, next);
+    result.streams[result.streams.length-1].on('error', errHandler);
+    result.with(response).pipeline(() => {});
   } else if (result.pipe != null) {
-    pipeline(result, response, rejectIfError(streamErrorHandler(response, next)));
+    const errHandler = streamErrorHandler(response, next);
+    result.on('error', errHandler);
+    pipeline(result, response, () => {});
```

Comment:
The docs for `pipeline()` don't state that the argument is optional: https://nodejs.org/docs/latest-v18.x/api/stream.html#streampipelinesource-transforms-destination-callback

It may be that the actual error handler should be `err => next(err)`, and `next` should be removed from the `streamErrorHandler()` function.